include/psa/crypto_struct.h

```diff
@@ -223,6 +223,29 @@ static inline struct psa_key_derivation_s psa_key_derivation_operation_init(
     return v;
 }
 
+struct psa_key_generation_method_s {
+    /* Future versions may add other fields in this structure. */
+    uint32_t flags;
+    uint8_t data[];
+};
+
+/** The default method for key generation or key derivation.
+ *
+ * Calling psa_generate_key_ext() or psa_key_derivation_output_key_ext()
+ * with `method=PSA_KEY_GENERATION_METHOD_INIT` and
+ * `method_data_length == 0` is equivalent to
+ * calling psa_generate_key() or psa_key_derivation_output_key()
+ * respectively.
+ */
+#define PSA_KEY_GENERATION_METHOD_INIT { 0 }
+
+static inline struct psa_key_generation_method_s psa_key_generation_method_init(
```

Comment:
GCC emits an annoying note about this function:
```
In file included from include/psa/crypto.h:4820,
                 from <stdin>:1:
include/psa/crypto_struct.h: In function ‘psa_key_generation_method_init’:
include/psa/crypto_struct.h:244:1: note: the ABI of passing struct with a flexible array member has changed in GCC 4.4
  244 | {
      | ^
```

We don't care about such old GCC versions. But the mere existence of this message is going to cost us in support requests. (Plus, it annoys me to see it every time.) I don't know how to silence it (AFAICT `#pragma GCC diagnostic` can't do it because it only affects messages controlled by some `-Wxxx`).

The assignment `method_lvalue = psa_key_generation_method_init()` is potentially misleading because it actually doesn't overwrite whatever is in `method.data`, which is potentially uninitialized memory. This is different from initialization, in that `psa_key_generation_method_s method_variable = PSA_KEY_GENERATION_METHOD_INIT;` creates a structure with empty `method.data`, so it does initialize everything.

@athoelke How do you feel about not having an initialization function for this structure, unlike the other structures? The anomaly being justified by having a flexible array member.